staging/src/k8s.io/api/resource/v1alpha3/types.go

```diff
@@ -1106,3 +1258,115 @@ type NetworkDeviceData struct {
+type DeviceTaint struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Spec specifies the filter and one taint.
+	//
+	// Changing the spec automatically increments the metadata.generation number.
+	Spec DeviceTaintSpec `json:"spec" protobuf:"bytes,2,name=spec"`
+
+	// ^^^
+	// A spec gets added because adding a status seems likely.
+	// Such a status could provide feedback on applying the
+	// eviction and/or statistics (number of matching devices,
+	// affected allocated claims, pods remaining to be evicted,
+	// etc.).
+}
+
+// DeviceTaintSpec specifies the filter and one taint.
+type DeviceTaintSpec struct {
+	// Filter defines which device(s) the patch is applied to.
+	// All filter criteria must be satified for a device to
+	// match. Without a filter or with an empty filter all
+	// devices match.
+	//
+	// +optional
+	Filter *DeviceTaintFilter `json:"filter,omitempty" protobuf:"bytes,1,opt,name=filter"`
+
+	// The taint that gets applied to matching devices.
+	//
+	// +required
+	Taint DeviceTaintAtom `json:"taints,omitempty" protobuf:"bytes,2,rep,name=taint"`
+}
+
+// DeviceTaintFilter defines which device(s) a [DevicePatch] applies to.
+type DeviceTaintFilter struct {
+	// If DeviceClassName is set, the selectors defined there must be
+	// satisfied by a device to be patched. This field corresponds
+	// to class.metadata.name.
+	//
+	// +optional
+	DeviceClassName *string `json:"deviceClassName,omitempty" protobuf:"bytes,1,opt,name=deviceClassName"`
+
+	// If driver is set, only devices from that driver are patched.
+	// This fields corresponds to slice.spec.driver.
+	//
+	// +optional
+	Driver *string `json:"driver,omitempty" protobuf:"bytes,2,opt,name=driver"`
+
+	// If pool is set, only devices in that pool are patched.
+	//
+	// Also setting the driver name may be useful to avoid
+	// ambiguity when different drivers use the same pool name,
+	// but this is not required because selecting pools from
+	// different drivers may also be useful, for example when
+	// drivers with node-local devices use the node name as
+	// their pool name.
+	//
+	// +optional
+	Pool *string `json:"pool,omitempty" protobuf:"bytes,3,opt,name=pool"`
+
+	// If device is set, only devices with that name are patched.
+	// This field corresponds to slice.spec.devices[].name.
+	//
+	// Setting also driver and pool may be required to avoid ambiguity,
+	// but is not required.
+	//
+	// +optional
+	Device *string `json:"device,omitempty" protobuf:"bytes,4,opt,name=device"`
+
+	// Selectors define criteria which must be satisfied by a
+	// device to be patched. All selectors must be satisfied.
+	//
+	// +optional
+	// +listType=atomic
+	Selectors []DeviceSelector `json:"selectors,omitempty" protobuf:"bytes,5,rep,name=selectors"`
+}
+
+// DeviceCapacity describes a quantity associated with a device.
+type DeviceCapacity struct {
```

Comment:
Could we omit this type for now, or is it needed for taints?

Reply:
No, we don't need it for v1alpha3. Will remove it.